### PR TITLE
CI: GHA Test

### PR DESCRIPTION
**Pull request by dgasmith**
## Description

Tries out the GHA CI pr integration and updates the status badge.

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by dgasmith** [comment]
Lint check isn't picking up configuration settings and failing. Very odd.

**Comment by dgasmith** [comment]
iSort isn't picking up `setup.cfg` config, but black is which is the important one. Skipping isort/autoflake for now.

We originally pushed back against the lint checker as we were worried that it would hinder submissions as too complex. As we now have the `make format` command this seems pretty straight forward these days and we can mark it as not required for PRs that we feel need to be merged over perfect linting. Any thoughts here?

File: .github/workflows/Lint.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - name: Python Setup
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Create Environment
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -e '.[lint]'
```
Comment (loriab) [inline .github/workflows/Lint.yml]: is this installing black, isort, autoflake from the `lint` section of setup.py?

Comment (dgasmith) [reply .github/workflows/Lint.yml]: Yes.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -7,4 +7,5 @@
 <!-- Provide a brief single sentence for the changelog. -->
 
 ## Status
+- [ ] `make format` run on the codebase (may have to `conda install black isort autoflake -c conda-forge` or `pip install black isort autoflake`)
```
Comment (loriab) [inline .github/PULL_REQUEST_TEMPLATE.md]: I took the liberty of adding this directly to your branch, so feel free to tweak/object.

**Comment by loriab** [comment]
oh, and there's a `.yapfignore` file that should probably go.

**Pull request merged**